addons/account_payment_term/tests/test_payment_term.py

```diff
@@ -55,4 +67,74 @@ def test_payment_term_days_end_of_month_on_the(self):
             special_case.invoice_date = '2023-12-12'
 
         expected_date_special_case = self.invoice.line_ids.filtered(lambda l: l.account_id == self.company_data['default_account_receivable']).mapped('date_maturity'),
-        self.assertEqual(expected_date_special_case[0], [fields.Date.from_string('2024-02-29')])
+        self.assertEqual(expected_date_special_case[0], [fields.Date.from_string('2024-01-31')])
+
+    def test_payment_term_days_end_of_month_nb_days_0(self):
+        """
+        This test will check that payment terms with a delay_type 'days_end_of_month_on_the'
+        in combination with nb_days works as expected
+
+        Invoice date = 2024-05-23
+        # case 1
+        'nb_days' = 0
+        `days_next_month` = 30
+            -> 2024-05-23 + 0 days = 2024-05-23
+            => `date_maturity` -> 2024-05-30
+        # case 2
+        'nb_days' = 0
+        `days_next_month` = 31
+            -> 2024-05-23 + 0 days = 2024-05-23
+            => `date_maturity` -> 2024-05-31
+        """
+        self.pay_term_days_end_of_month_30.line_ids.nb_days = 0
+        self.pay_term_days_end_of_month_31.line_ids.nb_days = 0
+        # case 1
+        with Form(self.invoice) as case_30:
```

Comment:
To be consistent with the docstring what do you think calling this case_1 and for the other case_2 ? and the comment would become useless :smile: